### DISCUSSION TITLE
Convert/serialize/marshal column type to richer type and back

### DISCUSSION THREAD

**Discussion by davetron5000** [jeremyevans/sequel#2258] [General]
I have scoured the docs and cannot find if there is a way to do this.  Essentially I want to have the attribute method of a model return a custom type based on the underlying database type and back again.

For example, suppose my `Widget` model has the `check_inventory_time_of_day` attribute, that stores a time of day in 24 hour time, like `"4:30"` I store that as a `text` in the database.  Suppose I have a class `TimeOfDay` that can interpret that value.

I'd like:

```ruby
Widget.first.check_inventory_time_of_day #=> <#TimeOfDay hour=4 minute=30>
widget.check_inventory_time_of_day = TimeOfDay.new(hour: 14, minute: 00)
widget.save #=> Writes "14:00" to the database
```

I can't find docs on how the attr readers/writers are created to know if they are safe to override and if they are all that needs overriding.

**Comment by jeremyevans** [comment]
You probably want to use the serialization plugin.

In pretty much all cases in Sequel, you can override the method, then call super to get the default behavior (methods defined automatically are generally defined in a module included in the class).

**Comment by davetron5000** [comment]
Ah, ok, that worked!

```ruby
class Widget # ...
  plugin :serialization
  serialize_attributes [
    :to_s.to_proc,
    TimeOfDay.method(:parse),
  ], :check_inventory_time_of_day

end
````
